graphql/database/invitedUsers/utils.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable import/prefer-default-export */
+// limit how many users can be created
+import moment from 'moment'
+import sgMail from '@sendgrid/mail'
+import InvitedUsersModel from './InvitedUsersModel'
+import UserModel from '../users/UserModel'
+import {
+  ADMIN_MANAGEMENT,
+  getPermissionsOverride,
+} from '../../utils/permissions-overrides'
+import { SENDGRID_API_KEY } from '../../config'
+
+export const verifyAndSendInvite = async (
+  userContext,
+  inviterId,
+  inviteEmail,
+  invitingUser,
+  inviterName,
+  inviterMessage
+) => {
+  const override = getPermissionsOverride(ADMIN_MANAGEMENT)
+  sgMail.setApiKey(SENDGRID_API_KEY)
+  const [doesUserAlreadyExist, hasUserBeenInvitedRecently] = await Promise.all([
+    UserModel.query(override, inviteEmail)
+      .usingIndex('UsersByEmail')
+      .execute(),
+    InvitedUsersModel.query(override, inviteEmail)
+      .usingIndex('InvitesByInvitedEmail')
+      .where('created')
+      .gte(
+        moment()
+          .subtract(30, 'days')
+          .toISOString()
+      )
+      .execute(),
+  ])
+  if (doesUserAlreadyExist.length) {
+    return { email: inviteEmail, error: 'user already exists' }
+  }
+  if (hasUserBeenInvitedRecently.length) {
+    return { email: inviteEmail, error: 'user has been invited recently' }
+  }
+  const msg = {
+    to: inviteEmail,
+    from: invitingUser.email,
+    templateId: 'd-69707bd6c49a444fa68a99505930f801',
+    dynamicTemplateData: {
+      name: inviterName,
```

Comment:
We need to URI encode the username in the URL param, or it will fail for some users. See [invite friends](https://github.com/gladly-team/tab/blob/master/web/src/js/components/Settings/Profile/InviteFriendComponent.js#L29) logic for an example.